dp_cluster_reg/registrationflow.py

```diff
@@ -0,0 +1,281 @@
+            ambari.enable_trusted_proxy_for_ranger()
+        if 'ATLAS' in dp.dependency_names() or dp.optional_dependency_names():
+            ambari.enable_trusted_proxy_for_atlas()
+        if 'BEACON' in dp.dependency_names() or dp.optional_dependency_names():
+            ambari.enable_trusted_proxy_for_beacon()
+
+        print 'Waiting for knox topologies to get activated. Sleeping for 10 seconds...'
+        time.sleep(10)
+        print 'Cluster changes are complete! Please log into Ambari, confirm the changes made to your cluster as part of this script and restart affected services.'
+        user.any_input()
+
+        if not dp.check_ambari(knox):
+            return 1
+
+        print 'Registering cluster to DataPlane...'
+        response = dp.register_ambari(ambari, knox, user)
+        return self.handle_registration_response(response)
+
+
+"""
+  CMRegistrationFlow : control Operation for Clouder Manager based clusters
+
+"""
+
+
+class CMRegistrationFlow(BaseRegistrationFlow):
+    def __init__(self):
+        self.provider = 'CM'
+        self.dp_instance = None
+
+    def execute(self):
+
+        self.dp_instance = self.get_dp_instance()
+        dp = self.dp_instance
+        if not dp.version.startswith("1.3"):
+            print(
+                "Registering CM Based cluster is not supported in DP %s" %
+                dp.version)
+            return 1
+        print BColors.BOLD + "\nTell me about Cloudera Manager Instance" + BColors.ENDC
+        cm = ClouderaManager(
+            user.url_input(
+                'CM URL', 'cm.url'), user.credential_input(
+                'CM admin', 'cm.admin'))
+
+        if not CMPrerequisites(cm).satisfied():
+            return 1
+
+        #
+        # If number of clusters managed by CM = 1 , The script will execute knox based flow
+        #
+        knox = None
+        clusters_to_register = []
+        clusters_resp_from_dp = []
+        if cm.total_clusters == 1:
+            active_cluster = cm.clusters[0]
+            print BColors.BOLD
+            print "Tell me about Knox Instance" + BColors.ENDC
+            knox = KnoxAdminApi(
+                user.url_input(
+                    'Knox Admin URL', 'knox_admin.url', default=str(
+                        active_cluster.knox_url())), user.credential_input(
+                    'Knox Admin', 'knox_admin.user'))
+            topology_util = CMTopologyUtil(cm, [])
+            topologies_to_deploy = [
+                TokenTopology(
+                    dp.public_key()), DpProxyTopologyForCM(
+                    cm, [], topology_util)]
+            for topology in topologies_to_deploy:
+                print 'Deploying Knox topology:', topology.name
+                topology.deploy(knox)
+            # communication check for
+            clusters_resp_from_dp = dp.check_cm(cm, knox)
+        elif cm.total_clusters > 1:
+            clusters_resp_from_dp = dp.check_cm(cm, None)
```

Comment:
This would be a reasonable case to add a comment specifying that we cannot handle Knox based setup for > 1 clusters. Can you please add this?